### DISCUSSION TITLE
DatePicker cannot set timezone

### DISCUSSION THREAD

**Discussion by tianhuil** [mantinedev/mantine#3307] [Feature requests]
If I have an app and uses a `<DatePicker/>`, I only care about the day, not the time of day.  Unfortunately, `<DatePicker/>` stores the local timezone into the date.  That is, entering "2020-01-02" in Berlin actually saves "2020-01-01T23:00:00.000Z", which, when viewed in the UK, is the day before: "2020-01-01".

The expected behavior is to save off the datetime in a consistent timezone (i.e. UTC) so that "2020-01-01" is always "2020-01-01T00:00:00.000Z".

**Proposed Fix:**
- Can we add an optional (default `false`) `useUTC` flag that automatically does this conversion?
- Better yet, can we make this default `true` (this would break backward compatibility but is probably what developers actually intended)?

**Comment by rtivital** [comment]
DatePicker returns Date object that user selects from the calendar. You can remove timezone information from this object on your side.

**Comment by tianhuil** [comment]
For posterity, here is the code that we use.  I would love if this were just implemented via `useUTC={true}`.

```ts
import { DatePicker, DatePickerProps } from '@mantine/dates'
import React from 'react'

/**
 * Takes any datetime date and returns the date (as datetime) in UTC
 * This removes time of day and timezone data regardless of the timezone data was entered (i.e. via DatePicker)
 * @param date: Date - input datetime which may have a time or timezone set
 */
export const toUTCDate = (date: Date | null | undefined): Date | undefined => {
  if (!date) return undefined
  return new Date(Date.UTC(date.getFullYear(), date.getMonth(), date.getDate()))
}

export function fromUTCDate(date: Date): Date
export function fromUTCDate(date: null | undefined): undefined
export function fromUTCDate(date: Date | null | undefined): Date | undefined
export function fromUTCDate(date: Date | null | undefined): Date | undefined {
  if (!date) return undefined
  // some dates were set in PT and ET midnight
  // this will still work for those dates because their midnights occur after GMT
  return new Date(date.getUTCFullYear(), date.getUTCMonth(), date.getUTCDate())
}

export const DatePickerUTC = React.forwardRef<HTMLInputElement, DatePickerProps>(
  ({ onChange, value, ...props }: DatePickerProps, ref) => {
    return (
      <DatePicker
        {...props}
        ref={ref}
        onChange={
          onChange &&
          ((v) => {
            onChange(toUTCDate(v) ?? null)
          })
        }
        value={fromUTCDate(value)}
      />
    )
  }
)
```

**Comment by yuki-js** [comment]
Unlike the pair of date and time, the date in solo is generally not associated with any time zone in particular. (e.g. "Let's meet on Jan 8" doesn't imply arrival at exactly Jan 8, 00:00 UTC)
So the output should be date string, instead of `Date` object that have time information. 
While we can adjust the timezone, but it's a matter of type impedance.

`<input type="date">` returns a string in `YYYY-MM-DD` format. I think Mantine should also follow HTML spec.
